### PR TITLE
Jenkins churned

### PR DESCRIPTION
**Pull request by mbogor**
#### What's this PR do?
creates a jenkins route to automate the churned users jobs
#### Where should the reviewer start?
www/jenkins and task/removechurned
#### How should this be manually tested?
curl post strings to jenkins/churned route (don't forget to include token in the query)
i included a sample curling request in the JIRA subtask 
the data you pass should be in the "number email number email" format since the real data will include a data of when the user was churned followed by their email.
comment out the actual deleting function in task/removechurned since you don't want to actually send delete requests to WhenIWork lines 45-48
#### What are the relevant tickets?
INT-320 INT-352

### CODE REVIEW

File: tasks/removeChurned.js [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+const express = require('express');
+const router = express.Router();
 
-  Then run: node console removeChurned go.
-**/
-module.exports.go = function () {
-  getUsersToClean()
+
+router.post('/', function(req, res){
+  var emails = req.body.split(' ').filter((item, n)=>{if (n%2!==0) return item});
```
Comment (jrauschenberg) [inline tasks/removeChurned.js]: "n" isn't a very clear variable name.

File: tasks/removeChurned.js [hunk 2]
```diff
@@ -35,12 +41,14 @@ function getUsersToClean() {
...
       });
       // delete the user and all user's shifts 
-      uidsToClean.forEach(function(id){
+      uidsToClean.forEach(function(id, index){
         api.delete(`users/${id}?delete_shifts=true`, function(res){
           if(res.error) CONSOLE_WITH_TIME(`CHURNED DELETION OF USER ${id} FAILED:`, res.error)
-          CONSOLE_WITH_TIME('Deleted user', id)
+          else CONSOLE_WITH_TIME('Deleted user', id)
         });
+        if(index==uidsToClean.length-1) resolve();
```
Comment (jrauschenberg) [inline tasks/removeChurned.js]: Why the double equals?

File: www/jenkins/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+const express   = require('express');
+const router = express.Router();
+const app = express();
+const bodyParser = require('body-parser');
```
Comment (jrauschenberg) [inline www/jenkins/index.js]: Are you using req.body in this file?

Comment (mbogor) [inline www/jenkins/index.js]: yes, like 8

File: tasks/removeChurned.js (hunk 2 above)
Comment (mbogor) [inline tasks/removeChurned.js]: fixed, thanks!

File: tasks/removeChurned.js (hunk 1 above)
Comment (mbogor) [inline tasks/removeChurned.js]: changed!

**Comment by tongxiang** [comment]
Two thoughts: 

1) if we're turning `removeChurned` into a route, we should take it out of the `tasks` directory
2) is there utility to keeping it as a task, as well? That way, we could run it without hitting it as a route?

**Comment by mbogor** [comment]
@tong i moved the file and added email notifications; is it good to merge?

**Comment by tongxiang** [comment]
👍

**Pull request merged**